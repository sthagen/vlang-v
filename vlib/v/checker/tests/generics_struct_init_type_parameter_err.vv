struct Response<U> {
	result U
}

pub fn send<A, B>(res A, b B) string {
	msg := Response<U>{
		result: res
	}
	println(b)
	return '$msg'
}

fn main() {
	mut ret := send(123, 22.2)
	println(ret)
}
