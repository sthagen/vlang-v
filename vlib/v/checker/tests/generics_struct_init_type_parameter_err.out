vlib/v/checker/tests/generics_struct_init_type_parameter_err.vv:6:9: error: generic struct init type parameter `U` must be within the parameters `(A,B)` of the current generic function
    4 |
    5 | pub fn send<A, B>(res A, b B) string {
    6 |     msg := Response<U>{
      |            ~~~~~~~~~~~~
    7 |         result: res
    8 |     }
vlib/v/checker/tests/generics_struct_init_type_parameter_err.vv:7:3: error: cannot assign to field `result`: expected `U`, not `int`
    5 | pub fn send<A, B>(res A, b B) string {
    6 |     msg := Response<U>{
    7 |         result: res
      |         ~~~~~~~~~~~
    8 |     }
    9 |     println(b)
